internal/framework/controller/predicate/service.go

```diff
@@ -67,6 +69,16 @@ func (ServicePortsChangedPredicate) Update(e event.UpdateEvent) bool {
 // GatewayServicePredicate implements predicate functions for this Pod's Service.
 type GatewayServicePredicate struct {
 	predicate.Funcs
+	NSName types.NamespacedName
+}
+
+// Create implements the default CreateEvent filter for the Gateway Service.
+func (gsp GatewayServicePredicate) Create(e event.CreateEvent) bool {
```

Comment:
I actually don't know if these Create/Delete/Generic functions really do anything...because the other Service predicate we have just allows all Services through.